codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocket08FrameDecoder.java

```diff
@@ -396,23 +398,34 @@ private void unmask(ByteBuf frame) {
 
         // Remark: & 0xFF is necessary because Java will do signed expansion from
         // byte to int which we don't want.
-        int intMask = ((maskingKey[0] & 0xFF) << 24)
-                    | ((maskingKey[1] & 0xFF) << 16)
-                    | ((maskingKey[2] & 0xFF) << 8)
-                    | (maskingKey[3] & 0xFF);
+        long longMask = ((long) maskingKey[0] & 0xff) << 56 |
+                        ((long) maskingKey[1] & 0xff) << 48 |
+                        ((long) maskingKey[2] & 0xff) << 40 |
+                        ((long) maskingKey[3] & 0xff) << 32 |
+                        ((long) maskingKey[0] & 0xff) << 24 |
+                        ((long) maskingKey[1] & 0xff) << 16 |
+                        ((long) maskingKey[2] & 0xff) <<  8 |
+                        (long)  maskingKey[3] & 0xff;
 
         // If the byte order of our buffers it little endian we have to bring our mask
         // into the same format, because getInt() and writeInt() will use a reversed byte order
         if (order == ByteOrder.LITTLE_ENDIAN) {
-            intMask = Integer.reverseBytes(intMask);
+            longMask = Long.reverseBytes(longMask);
         }
 
-        for (; i + 3 < end; i += 4) {
-            int unmasked = frame.getInt(i) ^ intMask;
-            frame.setInt(i, unmasked);
+        for (int lim = end - 7; i < lim; i += 8) {
+            long unmasked = frame.getLong(i) ^ longMask;
+            frame.setLong(i, unmasked);
         }
+
+        if (i < end - 3) {
+            frame.setInt(i, frame.getInt(i) ^ (int) (longMask >> 32));
```

Comment:
Is the `>> 32` necessary?

Reply:
No, we can remove it and just cast to int, because we use the same low and high 32 bits.

Reply:
Fixed